qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -77,15 +78,18 @@ def __init__(
                 on the method mentioned.
         """
         validate_min("threshold", threshold, 1e-15)
-        validate_min("delta", delta, 1e-5)
+
+        if isinstance(gradient, float):
+            # this scenario can only occur while using the deprecate_arguments wrapper which will
+            # move any argument supplied to delta into gradient.
+            gradient = Gradient(grad_method="fin_diff", epsilon=gradient)
 
         if gradient is None:
             gradient = Gradient(grad_method="param_shift")
```

Comment:
I must admit I don't quite understand how gradient could ever be None if delta is copied over it - in the future when delta is gone, but now...

How do I supply a gradient explicitly now - maybe the copy is done only if gradient is not explicitly provided, or delta provided explicitly - though if its the latter case that would be problematic if defaulted (I should look at the deprecate arguments decorator code I guess!).

Reply:
Yes, the copy is only done by the deprecation decorator if and only if the user provided a `delta` explicitly. Thus, `gradient is None` can still occur when the user has provided neither a custom `delta` nor a custom `gradient`.

This scenario is also covered by the unittests (the `test_default` case).

Reply:
Let me ask more specifically. Is the default behavior when delta is not supplied explicitly, the same as it was before. Ie if I ran it before with default delta do I get the same still result - it would seem in this case it would drop with gradient is None path right, and hence it seems is should/could be different as I see the find_diff path being equivalent to the old way.

Reply:
The previous default behavior remains unchanged: when a delta is supplied, a `fin_diff` gradient with the given delta will be used (as in the old days). However, the new default behavior uses param_shift instead which we found to be more performant.

Reply:
So basically anyone who never supplied delta, but used the provided default, will see a difference then. So if some one wants to reproduce something and was implicitly using defaulted delta=1 to get the same outcome they would have to change their code to explicitly provide delta=1.

What I was suggesting before, as a transition step it could use fin_diff, based on delta, with gradient defaulting to None. To use gradient one would explicitly provide it and we could suggest to use param shift which could become default once delta is removed.

If you feel how it is better I imagine the algo did not get wide usage before so perhaps any impact to the default being different than before won't have any real world impact. If someone is trying to say reproduce something from before maybe docs should say that to get the same prior default behavior you can supply delta=1 explicitly.

Reply:
The end result is not affected significantly so I do not think that this change in default has any real world impact. We can state in the docs how to reproduce the original default case. @fs1132429 could you please add a sentence to the docs explaining this?